opencog/moses/scoring/precision_bscore.cc

```diff
@@ -174,274 +178,416 @@ precision_bscore::precision_bscore(const CTable& ctable_,
-    // selectors that are exactly correct, even if they have a terrible
-    // activation.  It will patch these together in the ensemble.  If
-    // these perfect scorers were penalized, they'd be discarded in the
-    // metapop, and we don't want that.  The member "_return_weighted_score"
-    // is a synonym for "we're doing boosting now".
-    if (not _return_weighted_score) {
-        // For boolean tables, activation sum of true and false positives
-        // i.e. the sum of all positives.   For contin tables, the activation
-        // is likewise: the number of rows for which the combo tree returned
-        // true (positive).
-        score_t activation = active / _ctable_weight;
-        score_t activation_penalty = get_activation_penalty(activation);
-        bs.push_back(activation_penalty);
-
-        if (logger().is_fine_enabled()) {
-            score_t precision = 0.0;
-            if (0 < active) {
-                // See above for explanation for the extra 0.5
-                precision = sao / active + 0.5;
-            }
-            logger().fine("precision = %f  activation=%f  activation penalty=%e",
-                          precision, activation, activation_penalty);
-        }
+	score_t active = 0.0;  // total weight of active outputs by handle
+	score_t sao = 0.0;     // sum of all active outputs
 
-        // Add time dispersion penalty
-        if (_pressure > 0.0) {
-            score_t dispersion_penalty =
-                get_time_dispersion_penalty(ctable_res.ordered_by_time());
-            bs.push_back(dispersion_penalty);
-            logger().fine("dispersion_penalty = %f", dispersion_penalty);
-        }
-    }
+	atomese::Interpreter interpreter(moses::compressed_value_key);
 
-    return bs;
-}
+	const ValuePtr result = interpreter(handle);
+    auto link_result = LinkValueCast(result)->value();
+	map<TTable::value_type, Counter<bool, count_t>> time2res;
+	CTable ctable_res;
+    int i = 0;
+	for (const CTable::value_type& io_row : _wrk_ctable) {
 
-/// scorer, for use with individual combo trees
-behavioral_score precision_bscore::operator()(const combo_tree& tr) const
-{
-    interpreter_visitor iv(tr);
-    auto interpret_tr = boost::apply_visitor(iv);
-
-    std::function<bool(const multi_type_seq&)> selector;
-    selector = [&](const multi_type_seq& irow)->bool
-    {
-        return id::logical_true == interpret_tr(irow.get_variant());
-    };
-
-    behavioral_score bs(do_score(selector));
-    log_candidate_bscore(tr, bs);
-    return bs;
+		const multi_type_seq& irow = io_row.first;
+		const auto& orow = io_row.second;
+
+		score_t sumo = 0.0;  // (tp-fp)/2 if active, else 0 if not active
+		score_t acto = 0.0;  // fp + tp if active, else 0 if not active
+        if (HandleCast(link_result.at(i))->get_type() == TRUE_LINK){
+            sumo = sum_outputs(orow);
+            sao += sumo;
+            acto = orow.total_count();
+            active += acto;
+        }
```

Comment:
The if block above should be indented with tabs only.

Reply:
Also, why did you ported a specialization of `precision_bscore::do_score` for `Handle` instead of reusing `do_score`?